### PR TITLE
Misc small fixes

### PR DESCRIPTION
**Pull request by hepcat72**
## Summary Change Description

Miscellaneous small fixes to issues encountered while making changes to accommodate the data in the example studies, so their loads succeed.

Details:

- Removed a duplicate instance of `CompoundsLoader.DataHeaders.NAME` I encountered when simultaneously looking for an issue Michael reported while I was wouking on the contrib doc.  Incidentally, that issue turned out to be user error (me), but also inspired a new feature issue about preserving the Peak Group Conflicts dropdowns when they go through validation.
- Discovered that an exception can arise under the InfileError code when there is a percent sign in the error string.  Fixed one case.  There are likely potentially more, but the problematic feature is planned to be refactored.
- Added a pylint disable that arises in MaintainedModel when using newer linter versions, while I was working on the linter section of the contrib doc.
- Added a suggestion to the `UnexpectedLabels` error in the `PeakAnnotationsLoader` to make it clearer what the problem is.
- Updated/improved some comments in the `MSRunsLoader`.

## Affected Issues/Pull Requests

- Resolves no documented issue
- Merges into PR #1359

## Review Notes

See comments in-line.

## Checklist

This pull request will be merged once the following requirements are met.  The
author and/or reviewers should uncheck any unmet requirements:

- Review requirements
  - Minimum approvals: 1 
  - No changes requested
  - All blocking issues resolved by reviewers
  - Specific reviewers: @__add_username_here__
    
  - Review period: 2 days 
- Associated issue/pull request requirements:
  
  - [x] All requirements in affected issues marked "resolved" are satisfied
  - [x] All required pull requests are merged *(or none)*
- Basic requirements
  
  - [x] [All linters pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#linting)
  - [x] [All tests pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#quality-control)
  - [x] All conflicts resolved
- Overhead requirements
  
  - [x] [New/changed method tests implemented/updated *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#test-implementation)
  - [ ] [Updated *Unreleased* section of `changelog.md` *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/changelog.md)
  - [x] [Migrations created & committed *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#migration-process)

### CODE REVIEW

File: DataRepo/loaders/peak_annotations_loader.py [hunk 1]
```diff
@@ -1458,7 +1458,16 @@ def get_label_observations(self, row, pgrec: Optional[PeakGroup]):
...
+            suggestion = None
+            if pgrec is not None:
+                suggestion = (
+                    f"Check to make sure animal '{pgrec.msrun_sample.sample.animal.name}' has the correct "
+                    "tracer(s)."
+                )
+            olnp.set_formatted_message(
+                **infile_err_args,
+                suggestion=suggestion,
+            )
```
Comment (hepcat72) [inline DataRepo/loaders/peak_annotations_loader.py:1470]: C. This just adds a suggested resolution to the error (check the animal's tracers).

File: DataRepo/models/maintained_model.py [hunk 2]
```diff
@@ -720,7 +720,9 @@ def from_db(cls, *args, **kwargs):
                 f"Triggering lazy auto-update of fields: {cls.__name__}.{{{cs.join(lazy_update_fields)}}}"
             )
             # Trigger an auto-update
-            rec.save(fields_to_autoupdate=lazy_update_fields, via_query=True)
+            rec.save(  # pylint: disable=unexpected-keyword-arg
+                fields_to_autoupdate=lazy_update_fields, via_query=True
+            )
```
Comment (hepcat72) [inline DataRepo/models/maintained_model.py:725]: C. Newer linters than our current process (superlinter) uses, complains about this one, but the complaint is erroneous, IMO. There's an open issue about it in the mypy repo.

File: DataRepo/utils/exceptions.py [hunk 3]
```diff
@@ -204,7 +204,12 @@ def set_formatted_message(
             insertions = [vdict[k] for k in order]
             message = message % tuple(insertions)
         else:
-            message = message % self.loc
+            try:
+                message = message % self.loc
+            except TypeError:
+                if not message.endswith("\n"):
+                    message += "  "
+                message += "Location: %s."
```
Comment (hepcat72) [inline DataRepo/utils/exceptions.py:212]: C. This mitigates an exception when there's an extra unanticipated percent sign in the errors.

File: DataRepo/utils/exceptions.py [hunk 4]
```diff
@@ -3935,7 +3940,7 @@ class InfileDatabaseError(InfileError):
     def __init__(self, exception, rec_dict, **kwargs):
         if rec_dict is not None:
             nltab = "\n\t"
-            deets = [f"{k}: {v}" for k, v in rec_dict.items()]
+            deets = [f"{k}: {str(v).replace('%', '%%')}" for k, v in rec_dict.items()]
```
Comment (hepcat72) [inline DataRepo/utils/exceptions.py:3943]: C. This avoids the problematic percent sign that was mitigated above.

File: DataRepo/views/upload/submission.py [hunk 5]
```diff
@@ -171,7 +171,6 @@ def __init__(self):
             CompoundsLoader.DataSheetName: [
                 CompoundsLoader.DataHeaders.NAME,  # Consumes
                 CompoundsLoader.DataHeaders.FORMULA,  # Consumes
-                CompoundsLoader.DataHeaders.NAME,  # Consumes
```
Comment (hepcat72) [inline DataRepo/views/upload/submission.py:174]: C. Overlooked duplicate.

**Review by lsirse-devs — APPROVED** [review]

**Pull request merged**